akita/src/plugins/filters/filters-plugin.ts

```diff
@@ -0,0 +1,166 @@
+import { AddOptions, EntityState, ID } from '../../api/types';
+import { EntityCollectionPlugin } from '../entity-collection-plugin';
+import { FilterPlugin } from './filter-plugin';
+import { QueryEntity, SelectOptions } from '../../api/query-entity';
+import { createFilter, Filter, FiltersState, FiltersStore } from './filters-store';
+import { FiltersQuery } from './filters-query';
+import { combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { SortByOptions } from '../../api/query-config';
+import { isFunction } from '../../internal/utils';
+import { compareValues } from '../../internal/sort';
+
+export type FiltersParams = {
+  entityIds?: ID | ID[];
+  [key: string]: any;
+};
+
+/**
+ * Helper function to do search
+ * @param searchKey
+ * @param inObj
+ */
+export var searchObjFunction = function(searchKey: string, inObj) {
+  return Object.keys(inObj).some(function(key) {
+    return  typeof inObj[key] === 'string' && inObj[key].toLocaleLowerCase().includes(searchKey.toLocaleLowerCase());
+  });
+};
+
+/**
+ * Filters plugins is usefull to define some filters to be apply for your query
+ *
+ */
+export class FiltersPlugin<S extends EntityState<E> = any, E = any, P extends FilterPlugin<E, S> = any> extends EntityCollectionPlugin<E, P> {
+  private _selectFilter$: Observable<Filter[]>;
+  private readonly _filterStore: FiltersStore;
+  private readonly _filterQuery: FiltersQuery;
+
+  get filterStore(): FiltersStore {
+    return this._filterStore;
+  }
+
+  get filterQuery(): FiltersQuery {
+    return this._filterQuery;
+  }
+
+
+
+  constructor(protected query: QueryEntity<S, E>, private params: FiltersParams = {}) {
```

Comment:
In a second thought, we must ask the user to provide us with the FiltersStore name as you can have multiple plugins. If the user doesn't provide a custom name, we can take the default Store name and set the name to `${this.store.name}Filters`